taichi/backends/opengl/opengl_api.cpp

```diff
@@ -373,13 +395,21 @@ struct CompiledProgram::Impl {
           ctx.args[i] = accum_size;
           accum_size += size;
         }  // concat all extptr into my baseptr
-        TI_INFO("baseptr = {}", baseptr);
         iov.push_back(IOV{baseptr, accum_size});
       }
     }
-    for (const auto &ker : kernels) {
+    {
       auto guard = launcher->create_launch_guard(iov);
-      ker->launch();
+      for (const auto &ker : kernels) {
+        if (ker->rse.has_value()) {
+          auto gtm_now = guard.map_buffer(1); // TODO: RAII
```

Comment:
nit: usually i would write `auto*`. Since although we don't care about its type, it's still good to know that this is a pointer type.

Also, does `1` in `map_buffer()` means the global tmp buffer's index is `1`? If so, i hope we can have a constant defined for that, e.g. `kGlobalTmpBufferIndex = 1`. Magic numbers could quickly go confusing..

Reply:
Yes, it's actually the second argument pushed into iov:
https://github.com/taichi-dev/taichi/blob/73fe7c613c8a02959ed1f95337271edf944b9333/taichi/backends/opengl/opengl_api.cpp#L344-L348

This magic `1` is not used elsewhere, let's just add a commit to claim this fact.